Signal/src/view controllers/MessagesViewController.m

```diff
@@ -202,7 +232,15 @@ -(void) initializeTextView {
     [self.inputToolbar.contentView.textView  setFont:[UIFont ows_regularFontWithSize:17.f]];
     self.inputToolbar.contentView.leftBarButtonItem = _attachButton;
     
-    self.inputToolbar.contentView.rightBarButtonItem = _messageButton;
+    _recordRecognizer = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(recording:)];
+    [_recordButton addGestureRecognizer:_recordRecognizer];
+}
+
+- (NSString *) applicationDocumentsDirectory
```

Comment:
@jackflips so by default the "mic" button is shown which you can use to start recording, if you input text in the text field it switches to a "send" button?